homeassistant/components/reolink/light.py

```diff
@@ -140,11 +145,19 @@ async def async_turn_on(self, **kwargs: Any) -> None:
             brightness := kwargs.get(ATTR_BRIGHTNESS)
         ) is not None and self.entity_description.set_brightness_fn is not None:
             brightness_pct = int(brightness / 255.0 * 100)
-            await self.entity_description.set_brightness_fn(
-                self._host.api, self._channel, brightness_pct
+            try:
+                await self.entity_description.set_brightness_fn(
+                    self._host.api, self._channel, brightness_pct
+                )
+            except InvalidParameterError as err:
+                raise ServiceValidationError(err) from err
```

Comment:
Should also be `HomeAssistantError`

Reply:
Are you sure, I know a few months ago that was the case, but since then the dev docs now state to use ServiceValidationError: https://developers.home-assistant.io/docs/core/platform/raising_exceptions/

So in that case we should adjust the dev docs back to ValueError.

Now the dev docs state:

> Raise ServiceValidationError on an invalid user input

Reply:
Funny. Too new so I haven't even had time to see it yet 😆